darts/models/forecasting/regression_model.py

```diff
@@ -174,18 +183,18 @@ def __init__(
 
         for _lags, lags_name in lags_type_checks:
             raise_if_not(
-                isinstance(_lags, (int, list)) or _lags is None,
-                f"`{lags_name}` must be of type int or list. Given: {type(_lags)}.",
+                isinstance(_lags, (int, list, dict)) or _lags is None,
+                f"`{lags_name}` must be of type int, list or dict. Given: {type(_lags)}.",
             )
             raise_if(
                 isinstance(_lags, bool),
-                f"`{lags_name}` must be of type int or list, not bool.",
+                f"`{lags_name}` must be of type int, list or dict, not bool.",
```

Comment:
There is kind of a two step process where the types of the lags is first checked (lines 183 to 222) and then, based on the type, they are converted. I guess that it was to avoid processing all of them before realizing that the last lag has the wrong type.

I don't mind running the type check while converting the arguments.